lib/smack/SmackModuleGenerator.cpp

```diff
@@ -15,6 +15,11 @@ void SmackModuleGenerator::generateProgram(llvm::Module& m) {
 
   Naming naming;
   SmackRep rep(&getAnalysis<DSAAliasAnalysis>(), naming, program);
+  SmackBV* bv = &getAnalysis<SmackBV>();
+  if (SmackOptions::BitVectors || bv->isBVRequired())
```

Comment:
Hi Mike, 

Thanks a lot for your time for the code review.

For this functionality, a single pass as part of smack program has a problem. It is that we need to handle macro definitions in smack.h. To be more specific, many functions in smack.h has bit-vector versions and non-bit-vector ones since Boogie does not allow overloading. To control this, I use a macro called BITVECTOR which is pretty similar to MEMORY_MODEL_* macros. So when IR code is being generated, a flag must be set up which a llvm pass as part of smack cannot do. As a result, I was wondering if we could compile the pass as a individual binary from smack and add it to scripts.